lib/order.rb

```diff
@@ -0,0 +1,70 @@
+require_relative 'menu'
+require 'twilio-ruby'
+
+class Order
+
+  attr_reader :basket, :menu
+
+  def initialize(menu = Menu.new)
+    @menu = menu
+    @basket = []
+    @total = 0
+    @confirmed = false
+    time = (Time.now+(60*60))
+    @confirmation_text = "Thank you! Your order was placed and will be delivered before #{time.strftime("%H")}:#{time.strftime("%M")}. You ordered:"
+  end
+
+  def add_item(item, quantity = 1)
+    check_exists(item)
+    @confirmation_text += " #{quantity} portion(s) of #{item},"
+    retrieve_price(item)
+    recalculate_total(quantity)
+    basket << {item: item, quantity: quantity}
```

Comment:
You've lost precision again here.  Both `check_exists` and `retrieve_price` fetch the dish object from the menu, but then you revert back to storing only the name.  If you store a reference to the dish here, your basket has access to everything it needs - names, prices and quantities.